src/ol/events.js

```diff
@@ -0,0 +1,344 @@
+ */
+ol.events.bindListener_ = function(listenerObj) {
+  var boundListener = function(evt) {
+    var rv = listenerObj.listener.call(listenerObj.bindTo, evt);
+    if (listenerObj.callOnce) {
+      ol.events.unlistenByKey(listenerObj);
+    }
+    return rv;
+  }
+  listenerObj.boundListener = boundListener;
+  return boundListener;
+};
+
+
+/**
+ * Finds the matching {@link ol.events.ListenerObjType} in the given listener
+ * array.
+ *
+ * @param {!Array<!ol.events.ListenerObjType>} listenerArray Array of listeners.
+ * @param {!Function} listener The listener function.
+ * @param {boolean} useCapture The capture flag for the listener.
+ * @param {Object=} opt_this The `this` value inside the listener.
+ * @param {boolean=} opt_remove Remove the found listener from the array.
+ * @return {ol.events.ListenerObjType|undefined} The matching listener.
+ * @private
+ */
+ol.events.findListener_ = function(
+    listenerArray, listener, useCapture, opt_this, opt_remove) {
+  var listenerObj;
+  for (var i = 0, ii = listenerArray.length; i < ii; ++i) {
+    listenerObj = listenerArray[i];
+    if (listenerObj.listener === listener &&
+        listenerObj.useCapture == useCapture &&
+        listenerObj.bindTo === opt_this) {
+      if (opt_remove) {
+        listenerArray.splice(i, 1);
+      }
+      return listenerObj;
+    }
+  }
+  return undefined;
+};
+
+
+/**
+ * @param {ol.events.EventTargetLike} target Target.
+ * @param {ol.events.EventType|string} type Type.
+ * @return {Array.<ol.events.ListenerObjType>|undefined} Listeners.
+ */
+ol.events.getListeners = function(target, type) {
+  var listenerMap = target[ol.events.LISTENER_MAP_PROP_];
+  return listenerMap ? listenerMap[type] : undefined;
+};
+
+
+/**
+ * Registers an event listener on an event target. Inspired by
+ * {@link https://google.github.io/closure-library/api/source/closure/goog/events/events.js.src.html}
+ *
+ * This function efficiently binds a `listener` to a `this` object, and returns
+ * a key for use with {@link ol.events.unlistenByKey}.
+ *
+ * @param {EventTarget|ol.events.EventTarget|
+ *     {removeEventListener: function(string, Function, boolean=)}} target
+ *     Event target.
+ * @param {ol.events.EventType|string|Array.<(ol.events.EventType|string)>} type
+ *     Event type.
+ * @param {ol.events.ListenerFunctionType} listener Listener.
+ * @param {boolean=} opt_useCapture Use capture. For listeners on an
+ *     {@link ol.events.EventTarget}, `true` simply means that the listener will
+ *     be called before already registered listeners. Default is false.
+ * @param {Object=} opt_this Object referenced by the `this` keyword in the
+ *     listener. Default is the `target`.
+ * @param {boolean=} opt_once If true, add the listener as one-off listener.
+ * @return {ol.events.Key} Unique key for the listener.
+ */
+ol.events.listen = function(
+    target, type, listener, opt_useCapture, opt_this, opt_once) {
```

Comment:
In all 100+ call sites for `ol.events.listen()`, the `opt_useCapture` argument is `false` or `undefined`.  In all but a handful of these sites, we call with `opt_this`.  It would make for slightly nicer code if `opt_this` were the 4th argument and `opt_useCapture` were last.  And since we never use it, I think we'd be fine removing `opt_useCapture` altogether.

I've pushed two commits to my [`no-use-capture` branch](https://github.com/ahocevar/ol3/compare/no-goog-events...tschaub:no-use-capture).  If you agree about reordering the arguments so `opt_useCapture` is last, you can cherry pick the first.  If you agree that we could get rid of `opt_useCapture` altogether, you can cherry pick both.